satpy/readers/hsaf_nc.py

```diff
@@ -0,0 +1,167 @@
+    over the Meteosat IODC (Indian Ocean) disk.
+
+Notes:
+  * Externally compressed files (.gz) and uncompressed files are handled
+    transparently via Satpys ``generic_open``, without creating temporary
+    decompressed files on disk.
+  * The reader uses area definitions specified in the associated YAML
+    configuration file to determine projection, coverage, and resolution.
+  * Provides access to the ``rain_rate`` and ``q_ind`` datasets (including
+    accumulated variants where available).
+"""
+
+import datetime as dt
+import logging
+
+import xarray as xr
+
+from satpy.area import get_area_def
+from satpy.readers.core.file_handlers import BaseFileHandler
+from satpy.readers.core.utils import generic_open
+from satpy.utils import get_legacy_chunk_size
+
+LOG = logging.getLogger(__name__)
+
+platform_translate = {"MSG1": "Meteosat-8",
+                      "MSG2": "Meteosat-9",
+                      "MSG3": "Meteosat-10",
+                      "MSG4": "Meteosat-11",
+                      }
+
+CHUNK_SIZE = get_legacy_chunk_size()
+
+class HSAFNCFileHandler(BaseFileHandler):
+    """Handle H SAF NetCDF files, with optional .gz external compression.
+
+    This file handler handles H SAF Instantaneous Rain Rate products which contain:
+    - rr or acc_rr: rain rate in mm/h (instantaneous) or mm (accumulated)
+    - qind: quality index in percent
+
+    The data is on a geostationary projection grid.
+    This file handler is tested with H SAF h60/h60b, h63 & h90 data in NetCDF
+    format.
+    """
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Opens the nc file and store the nc data."""
+        super().__init__(filename, filename_info, filetype_info)
+        self._area_name = None
+        self._lon_0 = None
+        # use generic_open contextmanager to handle compression transparently
+        self._cm = generic_open(filename, mode="rb", compression="infer")
+        # manually enter the context manager
+        fp = self._cm.__enter__()
+        # lazily load nc file through h5netcdf without forcing read
+        self._nc_data = xr.open_dataset(fp, engine="h5netcdf", chunks="auto")
+
+    def __del__(self):
+        """Instruct the context manager to clean up and close the file."""
+        try:
+            if hasattr(self, "_cm") and self._cm:
+                self._cm.__exit__(None, None, None)
+        except (AttributeError, RuntimeError, OSError, ValueError):
+            LOG.warning(f"An error occurred while cleaning up the file {self.filename}")
+
+    def _get_global_attributes(self):
+        """Create a dictionary of global attributes to be added to all datasets."""
+        attributes = {
+            "filename": self.filename,
+            "spacecraft_name": platform_translate.get(self._nc_data.attrs["satellite_identifier"], "N/A"),
+            "platform_name": platform_translate.get(self._nc_data.attrs["satellite_identifier"], "N/A"),
```

Comment:
is `N/A` an acceptable value for these attributes? Does it ever happen that `satellite_identifier` is not defined?
Also, why do we need both `spacecraft_name` and `platform_name`?